### PR TITLE
feat (search): Add search capability for customers and plans

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Add search capability from the UI for customers and plans

## Description

Besides adding search capability, we started refactoring of our query logic. There is new BaseQuery class which is parent to all other query classes (e.g. CustomerQuery). This PR only handles Plans and Customers. Later PRs will cover other collections

### CODE REVIEW

File: app/services/queries/customer_query.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      result.customers = customers
+      result
+    end
+
+    private
+
+    attr_reader :search_term
+
+    def base_scope
+      Customer.where(organization: organization).ransack(search_params)
```
Comment (jdenquin) [inline app/services/queries/customer_query.rb:21]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: app/services/queries/plan_query.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      result.plans = plans
+      result
+    end
+
+    private
+
+    attr_reader :search_term
+
+    def base_scope
+      Plan.where(organization: organization).ransack(search_params)
```
Comment (jdenquin) [inline app/services/queries/plan_query.rb:21]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/customer_query_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+require 'rails_helper'
+
+RSpec.describe Queries::CustomerQuery, type: :service do
+  subject(:customer_query) do
+    described_class.new(organization: organization)
+  end
+
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:customer_first) { create(:customer, organization: organization, name: 'defgh') }
```
Comment (jdenquin) [inline spec/services/queries/customer_query_spec.rb:12]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/customer_query_spec.rb [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  let(:customer_second) { create(:customer, organization: organization, name: 'abcde') }
```
Comment (jdenquin) [inline spec/services/queries/customer_query_spec.rb:13]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/customer_query_spec.rb [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  let(:customer_third) { create(:customer, organization: organization, name: 'presuv') }
```
Comment (jdenquin) [inline spec/services/queries/customer_query_spec.rb:14]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/plan_query_spec.rb [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+require 'rails_helper'
+
+RSpec.describe Queries::PlanQuery, type: :service do
+  subject(:plan_query) do
+    described_class.new(organization: organization)
+  end
+
+  let(:membership) { create(:membership) }
+  let(:organization) { membership.organization }
+  let(:plan_first) { create(:plan, organization: organization, name: 'defgh') }
```
Comment (jdenquin) [inline spec/services/queries/plan_query_spec.rb:12]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/plan_query_spec.rb [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  let(:plan_second) { create(:plan, organization: organization, name: 'abcde') }
```
Comment (jdenquin) [inline spec/services/queries/plan_query_spec.rb:13]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/queries/plan_query_spec.rb [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  let(:plan_third) { create(:plan, organization: organization, name: 'presuv') }
```
Comment (jdenquin) [inline spec/services/queries/plan_query_spec.rb:14]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

**Review by vincent-pochet — APPROVED** [review]

**Review by ansmonjol — APPROVED** [review]

**Pull request merged**